packages/strapi-provider-upload-local/lib/index.js

```diff
@@ -12,19 +12,43 @@ module.exports = {
   provider: 'local',
   name: 'Local server',
   init: (config) => {
+    // write file in public/assets folder
+    const write = (file, dirPath) => {
+      return new Promise((resolve, reject) => {
+        fs.writeFile(path.join(dirPath, `${file.hash}${file.ext}`), file.buffer, (err) => {
+          if (err) {
+            return reject(err);
+          }
+
+          file.url = `/uploads/${file.hash}${file.ext}`;
+
+          resolve();
+        });
+      });
+    };
+
     return {
       upload: (file) => {
         return new Promise((resolve, reject) => {
-          // write file in public/assets folder
-          fs.writeFile(path.join(strapi.config.appPath, 'public', `uploads/${file.hash}${file.ext}`), file.buffer, (err) => {
-            if (err) {
-              return reject(err);
-            }
+          const dirPath = path.join(strapi.config.appPath, 'public', 'uploads');
```

Comment:
It should be done in the `bootstrap.js` file of the plugin... and not on every request.

see https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-upload/config/functions/bootstrap.js